### PR TITLE
Jetpack-connect: Check for jetpack versions

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR adds a minimum jetpack version required for jetpack-connect to work.

image

For any jetpack lesser than 3.9.6 we are going to show this message. We need to create some instructions to show the user about how to update.

How to test
========
1. ~~You need to have a .com sandbox running `D1581-code`~~ not anymore, already merged
2. Go to http://calypso.localhost:3000/jetpack/connect and enter the url of a jetpack site with a jetpack version smaller than 3.9.6
3. You should see a warning about your jetpack version

ping @roccotripaldi  @alternatekev

### CODE REVIEW

File: client/signup/jetpack-connect/index.jsx [hunk 1]
```diff
@@ -17,10 +17,13 @@ import Main from 'components/main';
 import JetpackConnectNotices from './jetpack-connect-notices';
 import SiteURLInput from './site-url-input';
 import { dismissUrl, goToRemoteAuth, goToPluginInstall, checkUrl } from 'state/jetpack-connect/actions';
+import versionCompare from 'lib/version-compare';
 
 const JetpackConnectMain = React.createClass( {
 	displayName: 'JetpackConnectSiteURLStep',
 
+	MINIMUM_JETPACK_VERSION: '3.9.6',
```
Comment (mtias) [inline client/signup/jetpack-connect/index.jsx]: I'd put this as a constant outside the class.

File: client/signup/jetpack-connect/index.jsx [hunk 2]
```diff
@@ -17,6 +17,13 @@ import Main from 'components/main';
 import JetpackConnectNotices from './jetpack-connect-notices';
 import SiteURLInput from './site-url-input';
 import { dismissUrl, goToRemoteAuth, goToPluginInstall, checkUrl } from 'state/jetpack-connect/actions';
+import versionCompare from 'lib/version-compare';
+
+
+/**
+ * Constants
+ */
+const MINIMUM_JETPACK_VERSION = '3.9.6';
```
Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: Needs to be `4.0.0`

**Comment by roccotripaldi** [comment]
Looks good. Other than the min version, i say ship it!

File: client/signup/jetpack-connect/index.jsx (hunk 2 above)
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: oh! Does the version in jetpack master's readme get bumped after the release? right now it marks 3.9.6 : https://github.com/Automattic/jetpack/blob/master/readme.txt#L4

Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: Yeah, it should get bumped. CC @eliorivero - make sure to bump version number:
https://github.com/Automattic/jetpack/blob/master/readme.txt#L4

**Comment by johnHackworth** [comment]
Ok, I'll leave this PR open until jetpack versions is bumped (I don't want to break our current local flow ;) )

File: client/signup/jetpack-connect/index.jsx (hunk 2 above)
Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: I was wrong about this. `3.9.6` is the correct stable tag to look for.

**Comment by roccotripaldi** [comment]
LGTM!

**Pull request merged**